tests/test_samples.py

```diff
@@ -1339,3 +1339,38 @@ def test_old_gui():
         make_2d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
     with pytest.raises(NotImplementedError):
         make_1d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
+
+
+def test_groupby_stats():
+    mcmc = read_chains('./tests/example_data/cb')
+    chains = mcmc.groupby('chain')
+    for chain in [1, 2]:
+        i = mcmc.chain == chain
+        assert_allclose(mcmc.loc[i].mean().drop('chain'),
+                        chains.mean().loc[chain, :])
+        assert_allclose(mcmc.loc[i].std().drop('chain'),
+                        chains.std().loc[chain, :])
+        assert_allclose(mcmc.loc[i].median().drop('chain'),
+                        chains.median().loc[chain, :])
+        assert_allclose(mcmc.loc[i].var().drop('chain'),
+                        chains.var().loc[chain, :])
```

Comment:
I think rather than using the `.loc[chain, :]` syntax it would be nicer to use the `chains.get_group(chain)` method, such that that gets tested as well.

Reply:
`chains.get_group(chain)` returns a `WeightedDataFrame`, so this change wouldn't actually be testing `WeightedDataFrameGroupBy`. Can additionally test this however

Reply:
Additional tests added

Reply:
hmm, I see what you are saying... but yes, we should have a test for `get_group` somewhere, too.

Reply:
resolved by @Ormorod in [5725ddf](https://github.com/williamjameshandley/anesthetic/pull/272/commits/5725ddfceb6d285c547cdbec8ed49f8596d2a9f8)